diff --git a/push/CMakeLists.txt b/push/CMakeLists.txt
index f6110b1..51cb281 100644
--- a/push/CMakeLists.txt
+++ b/push/CMakeLists.txt
@@ -1,5 +1,6 @@
 
 find_package(CURL REQUIRED)
+find_package(OpenSSL REQUIRED)
 
 add_library(push
   src/gateway.cc
@@ -21,6 +22,7 @@ target_link_libraries(push
   PUBLIC
     ${PROJECT_NAME}::core
     CURL::libcurl
+    OpenSSL::SSL OpenSSL::Crypto
   PRIVATE
     ${PROJECT_NAME}::util
     Threads::Threads
