diff --git a/test/cosigner/Makefile b/test/cosigner/Makefile
index a370b5e..a070f49 100644
--- a/test/cosigner/Makefile
+++ b/test/cosigner/Makefile
@@ -1,8 +1,8 @@
 App_Name := test
 
-App_C_Flags := -g -I../../include -DTEST_ONLY
+App_C_Flags := -g -I../../include -I$(BSPROOTFS)/include -DTEST_ONLY
 App_Cpp_Flags := $(App_C_Flags) -std=c++17
-App_Link_Flags := -L../../src/common -lcosigner -Wl,-Bstatic -lcrypto -Wl,-Bdynamic -luuid -lsecp256k1 -pthread -ldl
+App_Link_Flags := -L../../src/common -lcosigner -L$(BSPROOTFS)/lib -Wl,-Bstatic -lcrypto -Wl,-Bdynamic -luuid -lsecp256k1 -pthread -ldl
 
 App_Cpp_Files := $(wildcard ./*.cpp)
 App_Cpp_Objects := $(App_Cpp_Files:.cpp=.o)
