diff --git a/test/crypto/secp256k1_algebra/Makefile b/test/crypto/secp256k1_algebra/Makefile
index 93258af..1863325 100644
--- a/test/crypto/secp256k1_algebra/Makefile
+++ b/test/crypto/secp256k1_algebra/Makefile
@@ -1,6 +1,6 @@
 App_Name := test
 
-App_C_Flags := -g -Wall -Wextra -I../../../include
+App_C_Flags := -g -Wall -Wextra -I../../../include -I$(BSPROOTFS)/include
 App_Cpp_Flags := $(App_C_Flags) -std=c++17
 App_Link_Flags := -lcrypto
 
