diff --git a/include/tests/catch.hpp b/include/tests/catch.hpp
index eda7cad..fce43db 100644
--- a/include/tests/catch.hpp
+++ b/include/tests/catch.hpp
@@ -10825,7 +10825,7 @@ namespace Catch {
 
     // 32kb for the alternate stack seems to be sufficient. However, this value
     // is experimentally determined, so that's not guaranteed.
-    static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
+    static constexpr std::size_t sigStackSize = 32768;
 
     static SignalDefs signalDefs[] = {
         { SIGINT,  "SIGINT - Terminal interrupt signal" },
